cmd/system-probe/config/config.go

```diff
@@ -72,58 +73,40 @@ type Config struct {
 
 // New creates a config object for system-probe. It assumes no configuration has been loaded as this point.
 func New(configPath string) (*Config, error) {
-	aconfig.InitSystemProbeConfig(aconfig.Datadog)
-	aconfig.Datadog.SetConfigName("system-probe")
+	aconfig.SystemProbe.SetConfigName("system-probe")
 	// set the paths where a config file is expected
 	if len(configPath) != 0 {
 		// if the configuration file path was supplied on the command line,
 		// add that first so it's first in line
-		aconfig.Datadog.AddConfigPath(configPath)
+		aconfig.SystemProbe.AddConfigPath(configPath)
 		// If they set a config file directly, let's try to honor that
 		if strings.HasSuffix(configPath, ".yaml") {
-			aconfig.Datadog.SetConfigFile(configPath)
+			aconfig.SystemProbe.SetConfigFile(configPath)
 		}
 	}
-	aconfig.Datadog.AddConfigPath(defaultConfigDir)
-
-	_, err := aconfig.LoadWithoutSecret()
+	aconfig.SystemProbe.AddConfigPath(defaultConfigDir)
+	// load the configuration
+	_, err := config.LoadCustom(aconfig.SystemProbe, "system-probe", false)
+	// If `!failOnMissingFile`, do not issue an error if we cannot find the default config file.
 	var e viper.ConfigFileNotFoundError
-	if err != nil {
-		if errors.As(err, &e) || errors.Is(err, os.ErrNotExist) {
-			log.Infof("no config exists at %s, ignoring...", configPath)
+	if err != nil && (!errors.As(err, &e) || configPath != "") {
+		// special-case permission-denied with a clearer error message
+		if errors.Is(err, fs.ErrPermission) {
+			if runtime.GOOS == "windows" {
+				err = fmt.Errorf(`cannot access the system-probe config file (%w); try running the command in an Administrator shell"`, err)
```

Comment:
Do we still consider error wraps with `%w` useful?

Reply:
They are only useful is something higher in the call stack needs to check for a specific error. It is a reasonable default though.